drivers/abp2/abp2.c

```diff
@@ -0,0 +1,227 @@
+#else
+#pragma message("implement I2C code here")
+#endif
+    status = data_rx[0];
+    if (status & ABP2_STATUS_MASK) {
+        return ABP2_ERRSENS;
+    }
+
+    return res;
+}
+
+uint8_t abp2_getstatus(const abp2_t *dev)
+{
+    uint8_t status = 0xFF;      /* sensor status byte */
+    int res = ABP2_OK;
+
+#if defined(MODULE_ABP2_SPI)
+    res = spi_init_cs(dev->params->spi, dev->params->cs);
+#else
+#pragma message("implement I2C code here")
+#endif
+    if (res) {
+        return ABP2_ERRBUS;
+    }
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    status = spi_transfer_byte(dev->params->spi, dev->params->cs, false, ABP2_CMD_NOP);
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+
+    return status;
+}
+
+int abp2_read_raw(const abp2_t *dev, abp2_raw_t *raw_values)
+{
+    uint8_t status = 0xFF;      /* sensor status byte */
+    int res = ABP2_OK;
+    int len = sizeof(data_tx_nop_read) / sizeof(data_tx_nop_read[0]);
+    uint8_t data_rx[ABP2_RX_BUF_LEN];
+
+#if defined(MODULE_ABP2_SPI)
+    res = spi_init_cs(dev->params->spi, dev->params->cs);
+#else
+#pragma message("implement I2C code here")
+#endif
+    if (res) {
+        return ABP2_ERRBUS;
+    }
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    spi_transfer_bytes(dev->params->spi, dev->params->cs, false, data_tx_nop_read, data_rx, len);
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+    status = data_rx[0];
+    if (status & ABP2_STATUS_MASK) {
+        return ABP2_ERRSENS;
+    }
+    raw_values->cntPress = 0x00FFFFFF & (data_rx[1] << 16 | data_rx[2] << 8 | data_rx[3]);
+    raw_values->cntTemp = 0x00FFFFFF & (data_rx[4] << 16 | data_rx[5] << 8 | data_rx[6]);
+
+    return res;
+}
+
+int abp2_measure_read(const abp2_t *dev, abp2_raw_t *raw_values)
+{
+    uint8_t status = 0xFF;     /* sensor status byte */
+    int res = ABP2_OK;
+    int len = sizeof(data_tx_meas_read) / sizeof(data_tx_meas_read[0]);
+    uint8_t data_rx[ABP2_RX_BUF_LEN];
+
+#if defined(MODULE_ABP2_SPI)
+    res = spi_init_cs(dev->params->spi, dev->params->cs);
+#else
+#pragma message("implement I2C code here")
+#endif
+    if (res) {
+        return ABP2_ERRBUS;
+    }
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    spi_transfer_bytes(dev->params->spi, dev->params->cs, false, data_tx_meas_read, data_rx, len);
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+    status = data_rx[0];
+    if (status & ABP2_STATUS_MASK) {
+        return ABP2_ERRSENS;
+    }
+    raw_values->cntPress = 0x00FFFFFF & (data_rx[1] << 16 | data_rx[2] << 8 | data_rx[3]);
+    raw_values->cntTemp = 0x00FFFFFF & (data_rx[4] << 16 | data_rx[5] << 8 | data_rx[6]);
```

Comment:
Why do you need the bit mask? `data_rx` is `uint8_t` so you will never exceed `0x00FFFFFF` anyway.